cmd/bclient/bserver/bserver.go

```diff
@@ -0,0 +1,118 @@
+package bserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/antonholmquist/jason"
+	"github.com/ndlib/bendo/cmd/bclient/fileutil"
+	"sync"
+)
+
+var (
+	ItemFetchStatus error
+
+	fileIDMutex sync.Mutex
+
+	fileIDList []fileIDStruct
+
+	bendoServer *string
+	RemoteJason *jason.Object
+)
+
+type fileIDStruct struct {
+	fileid string
+	slot   string
+	item   string
+}
+
+func addFileToTransactionList(filename string, fileID string, item string) {
+
+	fileIDMutex.Lock()
+
+	thisFileID := new(fileIDStruct)
+	thisFileID.fileid = fileID
+	thisFileID.slot = filename
+	thisFileID.item = item
+
+	fileIDList = append(fileIDList, *thisFileID)
+
+	fileIDMutex.Unlock()
+}
+
+func Init(server *string) {
+	fileutil.IfVerbose("github.com/ndlib/bendo/bclient/bserver.Init() called")
+	bendoServer = server
+}
+
+func FetchItemInfo(item string) {
+	defer fileutil.UpLoadDone.Done()
+	fileutil.IfVerbose("github.com/ndlib/bendo/bclient/bserver.FetchItemInfo() called")
+
+	json, err := GetItemInfo(item)
+
+	// Some error occurred retrieving from server, or item not found.
+	if err != nil {
+		ItemFetchStatus = err
+		fmt.Println(err.Error())
+		return
+	}
+
+	RemoteJason = json
+}
+
+// serve the file queue. This is called from main as 1 or more goroutines
+// If the file Upload fails, close the channel and exit
+
+func SendFiles(fileQueue chan string, item string, fileroot string, mut *sync.WaitGroup) {
+
+	for filename := range fileQueue {
+		err := uploadFile(filename, fileutil.ShowUploadFileMd5(filename), item, fileroot)
+
+		if err != nil {
+			close(fileQueue)
```

Comment:
in general readers on a channel should not close it (since writes to a closed channel will cause a panic).

Also, it is nice to specify whether a channel is intended to be used as a reader or a writer in the function signature. In this case it is as a reader, so the type would be `fileQueue <-chan string`.